outlines/models/openai.py

```diff
@@ -243,6 +267,26 @@ def generate_choice(
 
         return choice
 
+    def generate_chat(
+        self, prompt: Union[str, List[str]], config: OpenAIConfig
+    ) -> np.ndarray:
+        """Call the async function to generate a chat response and keeps track of usage data.
+
+        Parameters
+        ----------
+        prompt
+            A string used to prompt the model as user message
+        config
+            An instance of `OpenAIConfig`.
+
+        """
+        results, usage = generate_chat(prompt, self.system_prompt, self.client, config)
+
+        self.last_usage = OpenAIUsage(**usage)
```

Comment:
I thought the idea of attaching the counts to the model class was good. Is there any reason why you changed your mind?